x-pack/test/functional/apps/maps/import_geojson/add_layer_import_panel.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import path from 'path';
+
+export default function ({ getPageObjects }) {
+  const PageObjects = getPageObjects(['maps', 'common']);
+
+  const IMPORT_FILE_PREVIEW_NAME = 'Import File';
+  const FILE_LOAD_DIR = 'test_upload_files';
+  const DEFAULT_LOAD_FILE_NAME = 'point.json';
+  const GEO_POINT = 'geo_point';
+  const GEO_SHAPE = 'geo_shape';
+
+  describe('Import layer panel', () => {
+    before(async () => {
+      await PageObjects.maps.openNewMap();
+    });
+
+    beforeEach(async () => {
+      await PageObjects.maps.clickAddLayer();
+      const panelOpen = await PageObjects.maps.isLayerAddPanelOpen();
+      expect(panelOpen).to.be(true);
+
+      await PageObjects.maps.selectGeoJsonUploadSource();
+      await PageObjects.maps.uploadJsonFileForIndexing(
+        path.join(__dirname, FILE_LOAD_DIR, DEFAULT_LOAD_FILE_NAME)
+      );
+      await PageObjects.maps.waitForLayersToLoad();
+      const layerLoadedInToc = await PageObjects.maps
```

Comment:
Move these two conditions to an `it` block. "Should add uploaded geojson file to map"

Reply:
Not sure I follow here. We need this to happen as part of `beforeEach`. We'll certainly still see if it fails since it's a requirement for all of the tests.

Reply:
Your beforeEach should not have any expect statements in it. It is just setting up the test. It does not verify anything. So these 2 expect statements need to be removed from the beforeEach.

If they are important, you can put them in an it block to make sure the beforeEach sets them up.